helper/controller/controller.go

```diff
@@ -149,44 +163,47 @@ func (c *Controller) processSingle(
 	informer cache.SharedIndexInformer,
 ) bool {
 	// Fetch the next item
-	key, quit := queue.Get()
+	rawEvent, quit := queue.Get()
 	if quit {
 		return false
 	}
-	defer queue.Done(key)
+	defer queue.Done(rawEvent)
 
-	// The key should be a string. If it isn't, just ignore it.
-	keyRaw, ok := key.(string)
+	event, ok := rawEvent.(Event)
 	if !ok {
-		c.Log.Warn("processSingle: dropping non-string key", "key", key)
+		c.Log.Warn("processSingle: dropping event with unexpected type", "event", rawEvent)
 		return true
 	}
 
-	// Get the item
-	item, exists, err := informer.GetIndexer().GetByKey(keyRaw)
+	// Get the item from the informer to ensure we have the most up-to-date
+	// copy.
+	key := event.Key
+	item, exists, err := informer.GetIndexer().GetByKey(key)
 
 	// If we got the item successfully, call the proper method
 	if err == nil {
-		c.Log.Debug("processing object", "key", keyRaw, "exists", exists)
+		c.Log.Debug("processing object", "key", key, "exists", exists)
 		c.Log.Trace("processing object", "object", item)
 		if !exists {
-			err = c.Resource.Delete(keyRaw)
+			// In the case of deletes, the item is no longer in the cache so
+			// we use the copy we got at the time of the event (event.Obj).
+			err = c.Resource.Delete(key, event.Obj)
```

Comment:
This is really the major change here. Previously we would just pass `key` into `Delete()`. Now we also pass `event.Obj` which contains the resource that was included with the delete event from kube. This is our only copy of the resource since `informer` has already had it cleared from its cache.